api/app/src/command/connected-user/get-connected-user.ts

```diff
@@ -34,10 +34,14 @@ export const getProviderDataFromConnectUserOrFail = (
   provider: ProviderOptions
 ) => {
   if (!connectedUser.providerMap) throw new NotFoundError("Could not find provider map");
-  const providerData = connectedUser.providerMap[provider];
-  if (!providerData) throw new NotFoundError("Provider map has no data");
 
-  return providerData;
+  if (connectedUser.providerMap[provider]?.token) {
+    const providerData = connectedUser.providerMap[provider];
+    if (!providerData) throw new NotFoundError("Provider map has no data");
+    return providerData;
+  } else {
+    throw new NotFoundError(`Could not find connect token for ${provider}`);
+  }
```

Comment:
Also, definitely NABD, but moving the `else` code out of the `if` usually makes the code easier to read - only one indentation/alternative flow to consider and keep track of. Not applicable in every situation, but it would work here.

But really NABD.

Reply:
Rookie mistake. I apologize for this :)

Reply:
Oh, please, no need to apologize! We're going through this together and benefit from external 👀 ... you'll see me 🤦'ing quite a lot! 😄